apps/evans/components/core/AudioPlayer.tsx

```diff
@@ -42,93 +37,101 @@ const AudioPlayer: React.FC<Props> = ({ tracks }) => {
 
   useEffect(() => {
     if (isPlaying) {
-      audioRef.current?.play();
+      audio?.play();
     } else {
-      audioRef.current?.pause();
+      audio?.pause();
     }
-  }, [isPlaying, currentTrackIndex.current]); // eslint-disable-line react-hooks/exhaustive-deps
+  }, [isPlaying, audio, currentTrackIndex]);
 
   useEffect(() => {
-    if (!audio) return;
-    audioRef.current?.addEventListener(`timeupdate`, () => {
-      setCurrentTime(audio.currentTime);
-    });
+    if (audio) {
+      audio.addEventListener(`timeupdate`, () => {
+        setCurrentTime(audio.currentTime);
+      });
+    }
   }, [audio]);
 
+  if (!track) return null;
+
   return (
     <div className="border border-flgray-200 bg-white p-4 min-[400px]:p-6 xs:p-8 flex flex-col">
-      <audio ref={audioRef} src={tracks[currentTrackIndex.current]?.url ?? ``} />
-      {audio && (
-        <div className={cx(`flex flex-col gap-4`, tracks.length > 1 && `mb-8`)}>
-          <div className="flex items-center justify-center md:justify-between">
-            <button
-              onClick={() => setIsPlaying(!isPlaying)}
-              className="w-12 h-12 bg-flprimary rounded-full flex justify-center items-center hover:opacity-90 transition-[opacity,transform] duration-200 active:scale-95 active:!opacity-100 shrink-0 order-1 md:order-0 mx-8 md:mx-0"
-            >
-              <i
-                className={cx(
-                  `fa-solid text-lg text-white`,
-                  isPlaying ? `fa-pause` : `fa-play`,
-                )}
-              />
-            </button>
-            <div className="flex-grow bg-flprimary/5 py-2 px-6 rounded-full flex justify-start overflow-hidden relative mx-8 hidden md:block md:order-1">
-              <span className="text-xl font-bold text-flprimary whitespace-nowrap">
-                {currentTitle}
-              </span>
-              <div className="absolute h-full w-24 right-0 top-0 bg-gradient-to-l from-[#F7F5F5] to-transparent" />
-            </div>
-            <button
-              onClick={() => {
-                setCurrentTime(currentTime - 10);
-                audio.currentTime = currentTime - 10;
+      <audio ref={audioRef} src={track.mp3Url} />
+      <div
+        className={cx(
+          `flex flex-col gap-4`,
+          tracks.length > 1 && `mb-8`,
+          !audio && `opacity-75 cursor-not-allowed [&>*]:pointer-events-none`,
```

Comment:
changed it so we always render the audio area, but on the very first render, when we don't have a ref yet, it's slightly grayed out and non-interactive